tests/unit/mixins/buffered-proxy-state-test.js

```diff
@@ -0,0 +1,101 @@
+import Ember from 'ember';
+import { module, test } from 'qunit';
+import StateMixin from 'ember-state-services/mixin';
+import BufferedProxy from 'ember-buffered-proxy/proxy';
+
+var State;
+var Service;
+var container;
+var subject;
+
+module('State Mixin - Buffer Proxy', {
+  setup: function() {
+    container = new Ember.Container();
+    State = BufferedProxy.extend();
+
+    container.register('state:test-state', State, {
+        singleton:   true,
+        instantiate: false
+    });
+
+    Service = Ember.Service.extend(StateMixin, {
+      stateName: 'test-state',
+      container: container
```

Comment:
we should pass this in to `Service.create` we don't want instance state to leak onto factories.